pywhat/distribution.py

```diff
@@ -0,0 +1,95 @@
+import json
+import os
+from typing import Optional
+
+from pywhat.helper import AvailableTags, InvalidTag
+
+
+class Distribution:
+    """
+    A distribution is an object containing the regex
+    But the regex has gone through a filter process.
+
+    Example filters:
+    * {"Tags": ["Networking"]}
+    * {"Tags": ["Identifiers"], "ExcludeTags": ["Credentials"], "MinRarity": 0.6}
+    """
+
+    def __init__(self, filters_dict: Optional[dict] = None):
+        tags = AvailableTags().get_tags()
+        self._dict = dict()
+        if filters_dict is None:
+            filters_dict = dict()

```

Comment:
We don't need this because of the function argument, no? We can change it:

```diff
- def __init__(self, filters_dict: Optional[dict] = None):
+ def __init__(self, filters_dict: Optional[dict] = {}):
```

Reply:
Yes, we can; however, I don't think it would be great since if somebody ever decides to change the code and mutate filters_dict, python will produce unexpected behaviour. So, in my opinion, it is better to write 2 extra lines of code rather than to be confused by the way python behaves if you ever decide to change the code. What do you think?